cmd/xl-v1.go

```diff
@@ -216,6 +227,13 @@ func (xl xlObjects) Shutdown() error {
 	return nil
 }
 
+// HealDiskMetadata function for object storage interface.
+func (xl xlObjects) HealDiskMetadata() error {
+	xl.diskMetadataLock.Lock()
+	defer xl.diskMetadataLock.Unlock()
```

Comment:
We should lock 'format.json' this is a distributed setup, simultaneously invoking the same 'control' heal would corrupt 'format.json'

```
      // Acquire a distributed lock to ensure only one of the nodes
                // initializes the format.json.
                nsMutex.Lock(minioMetaBucket, formatConfigFile)
                defer nsMutex.Unlock(minioMetaBucket, formatConfigFile)
```